### DISCUSSION TITLE
How to create horizontal well in GEOSX

### DISCUSSION THREAD

**Discussion by wdmhouston** [GEOS-DEV/GEOS#1452] [Q&A]
Hi,
 
     I am trying to make a horizontal well with the following polylineNodeCoords entry,
````
polylineNodeCoords="{ { 5025, 4525, 25},
                            { 5025, 5525, 25} }"
````
but this doesn't work.  So, what is the correct workflow to create a horizontal well in GEOSX?

**Comment by francoishamon** [comment]
You can create a horizontal well, but you need a first segment that is not completely horizontal. Something like that:
```
|
|
 \
   \           perf 1       perf 2     perf 3
      ------------'----------'-----------'-
```
so you can try:
```
polylineNodeCoords="{ { 5025, 4525, 50},,
                      { 5025, 4525, 25},
                      { 5025, 4550, 25} }"
polylineSegmentConn="{ { 0, 1 },
                       { 1, 2 } }"
```
That gives a well of length 25 + 25 = 50 meters
Then assume that you want to add a perforation in the middle of the horizontal segment, that is, at 25 + 12.5 meters of the well head along the well:
```
<Perforation
        name="injector1_perf1"
        distanceFromHead="37.5"/>
```

Now if you want to another horizontal segment, and another perforation:
```
polylineNodeCoords="{ { 5025, 4525, 50},
                      { 5025, 4525, 25},
                      { 5025, 4550, 25},
                      { 5025, 4575, 25} }"
polylineSegmentConn="{ { 0, 1 },
                       { 1, 2 }, 
                       { 2, 3 } }"
```
and 
```
<Perforation
        name="injector1_perf1"
        distanceFromHead="37.5"/>
<Perforation
        name="injector1_perf2"
        distanceFromHead="62.5"/>
```
Make sure that there is always a perforation in the bottom segment of the well. If you get convergence problems with the horizontal wells, let me know. I suggest starting the simulation with small time steps.

**Reply by wdmhouston** [reply]
Thanks! Will try!

**Reply by wdmhouston** [reply]
Francois,
  I'm trying to play horizontal well now.

for the example mentioned earlier,
````
polylineNodeCoords="{ { 5025, 4525, 50},
                      { 5025, 4525, 25},
                      { 5025, 4550, 25},
                      { 5025, 4575, 25} }"
polylineSegmentConn="{ { 0, 1 },
                       { 1, 2 }, 
                       { 2, 3 } }"
<Perforation
        name="injector1_perf1"
        distanceFromHead="37.5"/>
<Perforation
        name="injector1_perf2"
        distanceFromHead="62.5"/>
````
Does this mean injector1_perf1 is from 37.5 to 62.5, and injector1_perf2 is from 62.5 to 75 (measure depth from header (5025, 4525, 50))?

and if I want to disable a perforation, can I just set perforation transmissibility to 0?
